src/DI/ServiceLookup/ILEmit/ILEmitResolverBuilder.cs

```diff
@@ -0,0 +1,423 @@
+            }
+            else
+            {
+                // push length
+                argument.Generator.Emit(OpCodes.Ldc_I4, enumerableCallSite.ServiceCallSites.Length);
+                // new ItemType[length]
+                argument.Generator.Emit(OpCodes.Newarr, enumerableCallSite.ItemType);
+                for (int i = 0; i < enumerableCallSite.ServiceCallSites.Length; i++)
+                {
+                    // array
+                    argument.Generator.Emit(OpCodes.Dup);
+                    // push index
+                    argument.Generator.Emit(OpCodes.Ldc_I4, i);
+                    // create parameter
+                    VisitCallSite(enumerableCallSite.ServiceCallSites[i], argument);
+                    argument.Generator.Emit(OpCodes.Stelem, enumerableCallSite.ItemType);
+                }
+            }
+
+            return null;
+        }
+
+        protected override Expression VisitFactory(FactoryCallSite factoryCallSite, ILEmitResolverBuilderContext argument)
+        {
+            if (argument.Factories == null)
+            {
+                argument.Factories = new List<Func<IServiceProvider, object>>();
+            }
+
+            // this
+            argument.Generator.Emit(OpCodes.Ldarg_0);
+            //      .Factories
+            argument.Generator.Emit(OpCodes.Ldfld, FactoriesField);
+
+            //                 i
+            argument.Generator.Emit(OpCodes.Ldc_I4, argument.Factories.Count);
+            //                [ ]
+            argument.Generator.Emit(OpCodes.Ldelem, typeof(Func<IServiceProvider, object>));
+
+            //                       provider
+            argument.Generator.Emit(OpCodes.Ldarg_1);
+            //                     (         )
+            argument.Generator.Emit(OpCodes.Call, ExpressionResolverBuilder.InvokeFactoryMethodInfo);
+
+            argument.Factories.Add(factoryCallSite.Factory);
+            return null;
+        }
+
+
+        private void AddConstant(ILEmitResolverBuilderContext argument, object value)
+        {
+            if (argument.Constants == null)
+            {
+                argument.Constants = new List<object>();
+            }
+
+            argument.Generator.Emit(OpCodes.Ldarg_0);
+            argument.Generator.Emit(OpCodes.Ldfld, ConstantsField);
+
+            argument.Generator.Emit(OpCodes.Ldc_I4, argument.Constants.Count);
+            argument.Generator.Emit(OpCodes.Ldelem, typeof(object));
+            argument.Constants.Add(value);
+        }
+
+
+        private Func<ServiceProviderEngineScope, object> BuildType(IServiceCallSite callSite)
+        {
+            var dynamicMethod = new DynamicMethod("ResolveService", MethodAttributes.Public | MethodAttributes.Static, CallingConventions.Standard, typeof(object), new [] {typeof(ILEmitResolverBuilderRuntimeContext), typeof(ServiceProviderEngineScope) }, GetType(), true);
```

Comment:
This line is too long. Comment on why you had to pass true as the last argument and use the named parameter syntax.